crates/red_knot_python_semantic/src/types.rs

```diff
@@ -2490,130 +2444,370 @@ impl<'db> Type<'db> {
                             ]),
                             None,
                         ),
-                    ])
-                }
+                    ],
+                );
+                Signatures::single(signature)
+            }
 
-                let mut binding = bind_call(db, arguments, overloads(db), self);
-                let Some((_, overload)) = binding.matching_overload_mut() else {
-                    return Err(CallError::BindingError { binding });
-                };
+            Type::FunctionLiteral(function_type) => Signatures::single(CallableSignature::single(
+                self,
+                function_type.signature(db).clone(),
+            )),
+
+            Type::ClassLiteral(ClassLiteralType { class })
+                if class.is_known(db, KnownClass::Bool) =>
```

Comment:
we could probably micro-optimize this slightly by doing an inner `match` over `class.known()` again rather than lots of `Type::ClassLiteral` branches with `if class.is_known()` guards

Reply:
Done